### PR TITLE
Moved flaky universe integration test to regression test only covering past problem

### PR DESCRIPTION
**Pull request by unterstein**
Summary:
The system integration test to install the Neo4j universe package is flaky some times. For whatever reason it is pretty stable installing it manually, maybe it takes too long time to spin up and start properly. For this reason this PR will remove this system integration test and add a test which tests only the regression marathon had in March 2017. The commit fixed this behavior for the Marathon 1.4.1 release is https://github.com/mesosphere/marathon/commit/7e8cfa4949214198b2cef81c7e82212f2ff936a4. The initial problem was, that resident tasks with user network never became healthy. This new integration test only reflects this regression and keeps the other moving parts out of the test :)

There is a ticket to track efforts to refactor the Neo4j universe package here https://jira.mesosphere.com/browse/DCOS_OSS-2126

@jeschkies could you please start the integration test for this branch?
@kensipe could you please have a look at the retry magic? I used copied this part and not sure if it is needed/correct.

Thanks <3

### CODE REVIEW

File: tests/system/marathon_common_tests.py [hunk 1]
```diff
@@ -1007,6 +1007,32 @@ def test_metric_endpoint(marathon_service_name):
...
+    app = client.get_app(app_id)
+
+    assert len(tasks) == 1, "The number of tasks is {} after deployment, but only 1 was expected".format(len(tasks))
+    assert app['container']['type'] == 'DOCKER', "The container type is not DOCKER"
+    assert len(app['container']['volumes']) == 2, "The container does not have the correct amount of volumes"
+
+    # check if app becomes healthy
+    @retrying.retry(wait_fixed=1000, stop_max_attempt_number=30, retry_on_exception=common.ignore_exception)
+    def check_health():
+        app = client.get_app(app_def["id"])
```
Comment (zen-dog) [inline tests/system/marathon_common_tests.py]: `s/app_def["id"]/app_id`

File: tests/system/marathon_common_tests.py [hunk 2]
```diff
@@ -1007,6 +1007,32 @@ def test_metric_endpoint(marathon_service_name):
...
+
+    client = marathon.create_client()
+    client.add_app(app_def)
+    shakedown.deployment_wait(app_id=app_id)
+
+    tasks = client.get_tasks(app_id)
+    app = client.get_app(app_id)
+
+    assert len(tasks) == 1, "The number of tasks is {} after deployment, but only 1 was expected".format(len(tasks))
+    assert app['container']['type'] == 'DOCKER', "The container type is not DOCKER"
```
Comment (zen-dog) [inline tests/system/marathon_common_tests.py]: is this needed? how is supposed to be different from what's in the app definition? :)

Comment (unterstein) [reply tests/system/marathon_common_tests.py]: copied this from somewhere else, will remove it

File: tests/system/marathon_common_tests.py (hunk 1 above)
Comment (unterstein) [reply tests/system/marathon_common_tests.py]: ah thanks, also copy issue :p

File: tests/system/marathon_common_tests.py [hunk 3]
```diff
@@ -1007,6 +1007,31 @@ def test_metric_endpoint(marathon_service_name):
         "service.mesosphere.marathon.app.count is absent"
 
 
+def test_healtchcheck_and_volume():
```
Comment (zen-dog) [inline tests/system/marathon_common_tests.py]: Oh, and could you please put a comment here that's this is a regression test linking the original JIRA issue? Thx!

Comment (unterstein) [reply tests/system/marathon_common_tests.py]: I tried to find the original JIRA, but I could not find it :( Karsten raised the same question here https://phabricator.mesosphere.com/D612 but didn`t got an answer :(

The commit fixed the behavior was reviewed here https://phabricator.mesosphere.com/D606

Comment (zen-dog) [reply tests/system/marathon_common_tests.py]: a better name then? smth. that describes what is being tested e.g. `test_healthcheck_with_user_network_and_volumes`?

**Comment by zen-dog** [comment]
I let this PR run against a permissive cluster [here](https://jenkins.mesosphere.com/service/jenkins/view/Marathon/job/system-integration-tests/job/marathon-si-pr/112/) - failing `test_marathon_root.test_pod_with_container_bridge_network
` seems to be unrelated.

**Comment by jeschkies** [comment]
> I let this PR run against a permissive cluster here - failing `test_marathon_root.test_pod_with_container_bridge_network` seems to be unrelated.

Yep, that's a known bug.

**Comment by jeschkies** [comment]
Let's merge it once the SI test for DC/OS open is through.

**Comment by unterstein** [comment]
🎉

**Pull request merged**